PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py

```diff
@@ -424,6 +424,32 @@ def miniAOD_customizeCommon(process):
     delattr(process, 'selectedUpdatedPatJetsPuppiJetSpecific')
 
     task.add(process.slimmedJetsPuppi)
+
+    # Embed pixelClusterTagInfos in slimmedJets
+    process.patJets.addTagInfos = cms.bool(True)
+    process.patJets.tagInfoSources.append( cms.InputTag("pixelClusterTagInfos") )
+    process.slimmedJetsNoDeepFlavour.dropTagInfos = cms.string('0')
+    process.updatedPatJetsSlimmedDeepFlavour.addTagInfos = cms.bool(True)
+    process.updatedPatJetsSlimmedDeepFlavour.tagInfoSources.append( cms.InputTag("pixelClusterTagInfos") )
+    process.updatedPatJetsTransientCorrectedSlimmedDeepFlavour.addTagInfos = cms.bool(True)
+    process.updatedPatJetsTransientCorrectedSlimmedDeepFlavour.tagInfoSources = cms.VInputTag( cms.InputTag("pixelClusterTagInfos") )
+
+    # Exclude pixelClusterTagInfos from previous production (80XLegacy and 94XFall17)
+    from RecoBTag.Configuration.RecoBTag_cff import btagging, btaggingNoPixelCluster, pfBTaggingTask, pfBTaggingTaskNoPixelCluster
+    
+    from Configuration.Eras.Modifier_run2_miniAOD_80XLegacy_cff import run2_miniAOD_80XLegacy
+    run2_miniAOD_80XLegacy.toReplaceWith(pfBTaggingTask, pfBTaggingTaskNoPixelCluster)
+    run2_miniAOD_80XLegacy.toModify(process.patJets, addTagInfos = cms.bool(False) )
+    run2_miniAOD_80XLegacy.toModify(process.slimmedJetsNoDeepFlavour, dropTagInfos = cms.string('1') )
+    run2_miniAOD_80XLegacy.toModify(process.updatedPatJetsSlimmedDeepFlavour, addTagInfos = cms.bool(False) )
+    run2_miniAOD_80XLegacy.toModify(process.updatedPatJetsTransientCorrectedSlimmedDeepFlavour, addTagInfos = cms.bool(False) )
+    
+    from Configuration.Eras.Modifier_run2_miniAOD_94XFall17_cff import run2_miniAOD_94XFall17
+    run2_miniAOD_94XFall17.toReplaceWith(pfBTaggingTask, pfBTaggingTaskNoPixelCluster)
+    run2_miniAOD_94XFall17.toModify(process.patJets, addTagInfos = cms.bool(False) )
+    run2_miniAOD_94XFall17.toModify(process.slimmedJetsNoDeepFlavour, dropTagInfos = cms.string('1') )
```

Comment:
here and other parts of this file/pr
please drop type specifications for all parameters which already exist.
This is a safer way to protect from parameter name mistakes and will also help in possible parameter migrations.

Reply:
Type specifications have been dropped from this file (the other cfg being the pixelClusterTagInfos_cfi), except for InputTags.

Reply:
Type specifications have been dropped, except for TagInfos and VTagInfos.